python/GafferSceneUI/PrimitiveVariableQueryUI.py

```diff
@@ -0,0 +1,515 @@
+#
+#  Copyright (c) 2022, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import imath
+import functools
+import collections
+
+import IECore
+import IECoreScene
+
+import Gaffer
+import GafferUI
+import GafferScene
+import GafferSceneUI
+
+##########################################################################
+# Internal utilities
+##########################################################################
+
+def __getLabel( plug, parentPlug = None ) :
+
+	prefix = ""
+	suffix = ""
+
+	n = plug.node()
+	qPlug = n.queryPlug( plug )
+
+	prefix = qPlug["name"].getValue() if qPlug["name"].getValue() != "" else "none"
+
+	if parentPlug is not None :
+		oPlug = n.outPlugFromQuery( qPlug )
+
+		currentPlug = plug
+		while (
+			currentPlug is not None and
+			currentPlug != oPlug
+		) :
+
+			suffix = "." + currentPlug.getName() + suffix
+			currentPlug = currentPlug.parent()
+
+	result = prefix + suffix
+
+	return result
+
+##########################################################################
+# Output widget
+##########################################################################
+
+class _OutputWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, childPlug, **kw ) :
+
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Vertical, spacing = 4 )
```

Comment:
`self.__column = `, since this container stacks things vertically.

Reply:
done